src/Microsoft.ML.Data/DataLoadSave/Binary/BinaryLoader.cs

```diff
@@ -852,8 +850,11 @@ private BinaryLoader(IHost host, ModelLoadContext ctx, Stream stream)
                         _threads = Math.Max(1, Environment.ProcessorCount / 2);
                     }
 
-                    _generatedRowIndexName = ctx.LoadStringOrNull();
-                    ch.CheckDecode(_generatedRowIndexName == null || !string.IsNullOrWhiteSpace(_generatedRowIndexName));
+                    if (ctx.Header.ModelVerWritten == 0x00010002 || ctx.Header.ModelVerWritten == 0x00010003)
+                    {
+                        _generatedRowIndexName = ctx.LoadStringOrNull();
```

Comment:
>_generatedRowIndexName [](start = 24, length = 22)

This field is not needed anymore. You can load the string into a temporary variable, or even just call `ctx.LoadStringOrNull()` without assigning it.